### ISSUE TITLE
[4.0] New admin menu proposal (FLUX menu)

### ISSUE THREAD

**Issue by jonrz** [joomla/joomla-cms#15536]
I would like to propose a new menu layout for the admin area.
This is not only a proposal, but a work in progress, so input is appreciated, and I'll add your changes to the document if it fits with the idea.
Also, the document is editable, so if you want to make changes, either request permission (give me your gMail so I can allow you to edit), or make a copy, edit, and post here.

Please check the following document:
https://drive.google.com/open?id=1Z6wAhwW7RM_H5RX3Hg5_T0mpSXXdIL19eZH701R2r6k
This represents a basic layout and thorough organization for the new menu, and I spent almost no effort in designing it. I just made it decently presentable.
The final design should be refined by the Joomla Design Team.

This new menu organizes things in a more logical way, and present all of the related links in a single area, separated in visual blocks.
For example, all "content" related menu items are in the same panel. This eliminates the "where do I go to do this and that" we so frequently hear from new, and even intermediate, users.
This new menu intends to feel like a "dashboard", in a sense that the user will find everything he needs in a glimpse.

I intend to keep the code unified (desktop and mobile), with differences happening only at the CSS level.

As for branding, I understand that there is no full Joomla logo in view, and you might have an issue with that, but I have another proposal for a branded admin area, which should appease both integrators and corporate clients. I'm working with the JUX team to evolve this idea.

The tablet view will feature the full menu, since it uses less than half the screen.
The smartphone view will either use a scroll or an accordion, but I still need to do more research and experimentation in this area. Ideas are appreciated.

The "JET actions" (provisional name) you see there is an Idea I had during a talk with Cliff about the new Dashboard. It is a helper/wizard component that would streamline administration tasks, helping the rookie user climb the learning curve while actually getting things done.
Please, take a look at it here (work in progress):
https://drive.google.com/open?id=0B6QfGa3p1VQ7bl9vNHFFZExYVDg
This new component ideation is both a UX task and a development task, and I'll create a new issue to handle it in the near future.

**Comment by dgt41** [comment]
> For example, all "content" related menu items are in the same panel. This eliminates the "where do I go to do this and that" we so frequently hear from new, and even intermediate, users.

This is a great idea, you're not the first one that thought about this. BUT you need to make new hooks so installed components CAN append their relative items there. Content is not only Joomla's article, for some sites other components are the main area of interest. The problem is that I don't see how this can be done in J4, as this version is incremental and not a brand new concept (e.g. joomla x).

> This new menu intends to feel like a "dashboard", in a sense that the user will find everything he needs in a glimpse.

Personally (I'm a mac guy) I don't even like the idea, I like simple things and this is way too overcrowded (at least for my taste)

Here is an idea to remove the most rarely used menu entries: https://github.com/joomla/joomla-cms/issues/15451#issuecomment-296236789 without loosing any of the functionality (slightly different workflow)


So to recap here: try to get an answer from @wilsonge (he's the leader for J4) if the radical changes that you are proposing are in the limits of J4 and then either work on them or postpone this for Joomla X

**Comment by brianteeman** [comment]
> Content is not only Joomla's article, for some sites other components are the main area of interest. The problem is that I don't see how this can be done in J4, as this version is incremental and not a brand new concept (e.g. joomla x).

dont forget that the menu is no longer fixed in the hardcoded structure as seen prior to 3.7. It can now be completely customised

**Comment by jonrz** [comment]
@brianteeman Really? I'm glad to hear that.
Just to be clear, will it be possible to modify *any* menu?
Including "user", "content", etc?

**Comment by brianteeman** [comment]
as of today - yes

**Comment by dgt41** [comment]
@brianteeman still we are talking here about default db data, and also where the related new components will append their links. Also I mighty be wrong here...

**Comment by Bakual** [comment]
In his proposals the component links go to the "Components" submenu exactly like always.
With the custom admin menu, it can be moved however wherever the user wants it to be. There is no need for any hooks or special code imho (would be a bad idea anyway). It's also the same behavior with the current J4 menu, no difference.

**Comment by jonrz** [comment]
Yes, I took great care to avoid triggering core code changes.
This is a new information architecture and design, based on the current menu.
The idea is that this would be doable for J4, and even for J3.8 if time allows, so no structural changes.

The only elements there that could cause some trouble are the "Joomla up to date" and "components up to date" icons, which would trigger a query to several external sites on every page load, so we may leave those just at the dashboard. Note: I'm not a developer, so I might be squarely wrong here (maybe this info could be cached, or checked every x minutes independent of page change, I don't know).

**Comment by wilsonge** [comment]
I quite like this concept actually. It's not so different to what we have now so I'm assuming it can be done without huge amounts of extra work beyond @ciar4n 's work.

Can you do some mobile concepts though? I'm not quite sure how this is going to work on mobile (@C-Lodder made a good point here: https://github.com/joomla/joomla-cms/pull/15430#issuecomment-296956649)

**Comment by jonrz** [comment]
@wilsonge thanks for the positive answer.
I'll do the mobile concepts and update the document in a few days.
Cheers!

**Comment by C-Lodder** [comment]
What does "JET actions" do?

Even as an avid Joomla user, I have absolutely no idea what this menu item does, nor does the document explain this.

Still not liking this concept purely based on my comment linked by George

**Comment by ciar4n** [comment]
I like the structure change and how the menu items have been grouped. You appear to have found a place for everything that is easily understandable.

On the general design I tend to agree with @dgt41 (https://github.com/joomla/joomla-cms/pull/15430#issuecomment-296858021). I am not sure if splitting the menu in to a collection of 'modules' makes for a fluid and functional menu. I have no doubt that it could look better and may aid a complete novice however most Joomla users would be using this menu every day and should be almost able to navigate it with their eyes closed. That means a set vertical rhythm which splitting the menu into 'modules' will likely destroy.

The same options can be done in other ways (eg. inline). 

Your 'Content' tab...

![inline-options](https://cloud.githubusercontent.com/assets/2803503/25426592/952b0204-2a67-11e7-81c8-8a53330277cb.png)

**Comment by jonrz** [comment]
@ciar4n Hey, awesome idea!
Will implement and update the doc shortly.

**Comment by jonrz** [comment]
Page 4 in the doc was updated with @ciar4n design.
I`ll review the remaining pages and adjust accordingly.

**Comment by jonrz** [comment]
The document was updated with @ciar4n contribution.
It looks more concise now, with everything in a single column, except components, but that looks a little bit better now, and I can`t see a better solution for components right now.

@brianteeman After applying Ciaran`s idea, I can envision it on mobile quite easily.
Components will use an accordion system instead of two columns, and the + sign is quite universal (but it can also be a down arrow, and that will be the designer`s team choice, not mine).
Some other blocks will also use accordion, if there is not enough screen space (such as languages, extensions and users), with either a + sign or a down arrow.

**Comment by brianteeman** [comment]
I am closing this here as there is a duplicate but current issue directly in the ux tracker.
